cargo-dist/src/backend/installer/homebrew.rs

```diff
@@ -56,64 +40,87 @@ pub struct HomebrewInstallerInfo {
     pub install_libraries: Vec<LibraryStyle>,
 }
 
+/// All homebrew-specific fragments
+#[derive(Debug, Clone, Serialize)]
+pub struct HomebrewFragments<T> {
+    /// macOS AMD64 artifact
+    pub x86_64_macos: Option<T>,
+    /// macOS ARM64 artifact
+    pub arm64_macos: Option<T>,
+    /// Linux AMD64 artifact
+    pub x86_64_linux: Option<T>,
+    /// Linux ARM64 artifact
+    pub arm64_linux: Option<T>,
+}
+
 pub(crate) fn write_homebrew_formula(
     dist: &DistGraph,
-    source_info: &HomebrewInstallerInfo,
+    info: &HomebrewInstallerInfo,
+    fragments: &HomebrewFragments<ExecutableZipFragment>,
     manifest: &DistManifest,
 ) -> DistResult<()> {
-    let mut info = source_info.clone();
-
-    // Fetch any detected dependencies from the linkage data
-    // FIXME: ideally this would be writing to per-target dependencies,
-    // but for now we just shove them all into one big list.
-    use_linkage(manifest, &info.arm64_macos, &mut info.dependencies);
-    use_linkage(manifest, &info.x86_64_macos, &mut info.dependencies);
-    use_linkage(manifest, &info.arm64_linux, &mut info.dependencies);
-    use_linkage(manifest, &info.x86_64_linux, &mut info.dependencies);
-
-    // Grab checksums
-    use_sha256_checksum(manifest, &info.arm64_macos, &mut info.arm64_macos_sha256);
-    use_sha256_checksum(manifest, &info.x86_64_macos, &mut info.x86_64_macos_sha256);
-    use_sha256_checksum(manifest, &info.arm64_linux, &mut info.arm64_linux_sha256);
-    use_sha256_checksum(manifest, &info.x86_64_linux, &mut info.x86_64_linux_sha256);
+    let info = info.clone();
 
-    let script = dist
-        .templates
-        .render_file_to_clean_string(TEMPLATE_INSTALLER_RB, &info)?;
-    LocalAsset::write_new(&script, &info.inner.dest_path)?;
-    Ok(())
-}
-
-/// Grab the sha256 checksum for this artifact from the manifest
-fn use_sha256_checksum(
-    manifest: &DistManifest,
-    fragment: &Option<ExecutableZipFragment>,
-    checksum: &mut Option<String>,
-) {
     let checksum_key = ChecksumStyle::Sha256.ext();
-    if let Some(frag) = &fragment {
-        *checksum = manifest
+    let map_fragment = |fragment: ExecutableZipFragment| -> HomebrewFragment {
```

Comment:
Can we move this definition right above the only place we use it?

Reply:
In my mind, it already is — putting it inside the `macro_rules!` block simply makes the macro bigger and harder to review.